src/ConstructionBase.jl

```diff
@@ -55,7 +57,40 @@ function setproperties(obj; kw...)
 end
 
 setproperties(obj, patch::NamedTuple) = _setproperties(obj, patch)
+setproperties(obj::Tuple, patch::typeof(NamedTuple())) = obj
+function setproperties(obj::Tuple, patch::NamedTuple)
```

Comment:
Mhhh so this method throws an error anyway. Can you sketch a code snipped where you think `@noinline` here would be beneficial?

Reply:
I'm not sure it will be an issue here but in some cases a string interpolation macro can cause an allocation if it is inlined into parent code. Maybe that's historical now but I've worked on a bunch of codebases that avoided it.

See: https://github.com/JuliaLang/julia/issues/37639#issuecomment-771150651
And: https://github.com/JuliaLang/julia/blob/master/stdlib/SparseArrays/src/sparsematrix.jl#L55

Reply:
Ok thanks, I was not aware of that, I will do it.

Reply:
Tim holy does mention that the `@noinline` is legacy in that comment, so it might not be necessary anymore.